src/opencl/mod.rs

```diff
@@ -263,14 +372,24 @@ impl Program {
         })
     }
 
-    pub fn create_kernel(&self, name: &str, gws: usize, lws: usize) -> GPUResult<Kernel> {
+    /// Returns a kernel.
+    ///
+    /// The `global_work_size` does *not* follow the OpenCL definition. It is *not* the total
+    /// number of threads. Instead it follows CUDA's definition and is the number of
+    /// `local_work_size` sized thread groups.
+    pub fn create_kernel(
```

Comment:
best to add a note, how the local and global are used to calculate the total number in the opencl case